transport/src/main/java/io/netty/channel/oio/OioByteStreamChannel.java

```diff
@@ -126,25 +126,19 @@ protected void doWriteFileRegion(FileRegion region) throws Exception {
             outChannel = Channels.newChannel(os);
         }
 
-        long written = 0;
-        for (;;) {
-            long localWritten = region.transferTo(outChannel, written);
-            if (localWritten == -1) {
-                checkEOF(region);
+        long bytesToTransfer = obj.readableBytes();
+        long totalTransferred = 0;
+        while (obj.isReadable()) {
+            long transferred = obj.readTo(outChannel, obj.readableBytes());
+            if (transferred == -1) {
```

Comment:
Right, zero does not imply `EOF`. The javadoc for [`FileChannel.transferTo`](http://docs.oracle.com/javase/7/docs/api/java/nio/channels/FileChannel.html#transferTo(long,%20long,%20java.nio.channels.WritableByteChannel)) does not indicate that it will ever return `-1`.  The [`FileChannel.read`](http://docs.oracle.com/javase/7/docs/api/java/nio/channels/FileChannel.html#read(java.nio.ByteBuffer)), however does. I'm just wondering why we're checking for `-1` ... is it actually possible as a return value of `transferTo`, but just isn't documented?